<!DOCTYPE html>
<html lang='en'>
<head>
<title>0003-Fix-size-hints-under-Xwayland-scaling.patch - aur.git - AUR Package Repositories</title>
<meta name='generator' content='cgit v1.2.3-56-gcc08'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/static/css/cgit.css'/>
<script type='text/javascript' src='/cgit.js'></script>
<link rel='shortcut icon' href='/static/images/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='https://aur.archlinux.org/cgit/aur.git/atom/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git' type='application/atom+xml'/>
<link rel='vcs-git' href='https://aur.archlinux.org/wlroots-hidpi-xprop-git.git/' title='aur.git Git repository'/>
</head>
<body>
		<div id="archnavbar" class="anb-aur">
			<div id="archnavbarlogo"><h1><a href="/" title="Return to the main page">Arch Linux User Repository</a></h1></div>
			<div id="archnavbarmenu">
				<ul id="archnavbarlist">
					<li id="anb-home"><a href="https://www.archlinux.org/" title="Arch news, packages, projects and more">Home</a></li>
					<li id="anb-packages"><a href="https://www.archlinux.org/packages/" title="Arch Package Database">Packages</a></li>
					<li id="anb-forums"><a href="https://bbs.archlinux.org/" title="Community forums">Forums</a></li>
					<li id="anb-wiki"><a href="https://wiki.archlinux.org/" title="Community documentation">Wiki</a></li>
					<li id="anb-bugs"><a href="https://bugs.archlinux.org/" title="Report and track bugs">Bugs</a></li>
					<li id="anb-security"><a href="https://security.archlinux.org/" title="Arch Linux Security Tracker">Security</a></li>
					<li id="anb-aur"><a href="/" title="Arch Linux User Repository">AUR</a></li>
					<li id="anb-download"><a href="https://www.archlinux.org/download/" title="Get Arch Linux">Download</a></li>
				</ul>
			</div>
		</div><!-- #archnavbar -->
<div id='cgit'><table id='header'>
<tr>
<td class='main'>AUR : wlroots-hidpi-xprop-git.git</td></tr>
<tr><td class='sub'>AUR Package Repositories | click <a href='/pkgbase/wlroots-hidpi-xprop-git/'>here</a> to return to the package base details page</td><td class='sub right'></td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/aur.git/?h=wlroots-hidpi-xprop-git'>summary</a><a href='/cgit/aur.git/log/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git'>log</a><a class='active' href='/cgit/aur.git/tree/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git'>tree</a><a href='/cgit/aur.git/commit/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git'>commit</a><a href='/cgit/aur.git/diff/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git'>diff</a><a href='/cgit/aur.git/stats/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/aur.git/log/0003-Fix-size-hints-under-Xwayland-scaling.patch'>
<input type='hidden' name='h' value='wlroots-hidpi-xprop-git'/><select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='search' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/aur.git/tree/?h=wlroots-hidpi-xprop-git'>root</a>/<a href='/cgit/aur.git/tree/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git'>0003-Fix-size-hints-under-Xwayland-scaling.patch</a></div><div class='content'>blob: a63a619f1a6eecd41a7e0407296e7b1cd491d93c (<a href='/cgit/aur.git/plain/0003-Fix-size-hints-under-Xwayland-scaling.patch?h=wlroots-hidpi-xprop-git'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
</pre></td>
<td class='lines'><pre><code>From 7220722f9173a90479d81f4b359cf91956f59f28 Mon Sep 17 00:00:00 2001
From: Patrick Taibel &lt;git&#64;pmt97.com&gt;
Date: Sat, 12 Aug 2023 16:05:28 +0800
Subject: [PATCH 3/3] Fix size hints under Xwayland scaling

<span class="hl kwa">---</span>
 xwayland/xwm.c | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/xwayland/xwm.c b/xwayland/xwm.c
index 295f4d49..674a32e1 100644
<span class="hl kwa">--- a/xwayland/xwm.c</span>
<span class="hl kwb">+++ b/xwayland/xwm.c</span>
&#64;&#64; -729,17 +729,34 &#64;&#64; static void read_surface_normal_hints(struct wlr_xwm *xwm,
 		xsurface-&gt;size_hints-&gt;base_width = -1;
 		xsurface-&gt;size_hints-&gt;base_height = -1;
 	} else if (!has_base_size_hints) {
<span class="hl kwa">-		xsurface-&gt;size_hints-&gt;base_width = xsurface-&gt;size_hints-&gt;min_width;</span>
<span class="hl kwa">-		xsurface-&gt;size_hints-&gt;base_height = xsurface-&gt;size_hints-&gt;min_height;</span>
<span class="hl kwa">-	} else if (!has_min_size_hints) {</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;base_width = unscale(xwm, xsurface-&gt;size_hints-&gt;min_width);</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;base_height = unscale(xwm, xsurface-&gt;size_hints-&gt;min_height); </span>
 		xsurface-&gt;size_hints-&gt;min_width = xsurface-&gt;size_hints-&gt;base_width;
 		xsurface-&gt;size_hints-&gt;min_height = xsurface-&gt;size_hints-&gt;base_height;
<span class="hl kwb">+	} else if (!has_min_size_hints) {</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;min_width = unscale(xwm, xsurface-&gt;size_hints-&gt;base_width);</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;min_height = unscale(xwm, xsurface-&gt;size_hints-&gt;base_height);</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;base_width = xsurface-&gt;size_hints-&gt;min_width;</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;base_height = xsurface-&gt;size_hints-&gt;min_height;</span>
<span class="hl kwb">+	} else {</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;min_width = unscale(xwm, xsurface-&gt;size_hints-&gt;min_width);</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;min_height = unscale(xwm, xsurface-&gt;size_hints-&gt;min_height);</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;base_width = unscale(xwm, xsurface-&gt;size_hints-&gt;base_width);</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;base_height = unscale(xwm, xsurface-&gt;size_hints-&gt;base_height); </span>
 	}
 
 	if ((flags &amp; XCB_ICCCM_SIZE_HINT_P_MAX_SIZE) == 0) {
 		xsurface-&gt;size_hints-&gt;max_width = -1;
 		xsurface-&gt;size_hints-&gt;max_height = -1;
<span class="hl kwb">+	} else {</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;max_width = unscale(xwm, xsurface-&gt;size_hints-&gt;max_width);</span>
<span class="hl kwb">+		xsurface-&gt;size_hints-&gt;max_height = unscale(xwm, xsurface-&gt;size_hints-&gt;max_height);</span>
 	}
<span class="hl kwb">+</span>
<span class="hl kwb">+	xsurface-&gt;size_hints-&gt;x = unscale(xwm, xsurface-&gt;size_hints-&gt;x);</span>
<span class="hl kwb">+	xsurface-&gt;size_hints-&gt;y = unscale(xwm, xsurface-&gt;size_hints-&gt;y);</span>
<span class="hl kwb">+	xsurface-&gt;size_hints-&gt;width = unscale(xwm, xsurface-&gt;size_hints-&gt;width);</span>
<span class="hl kwb">+	xsurface-&gt;size_hints-&gt;height = unscale(xwm, xsurface-&gt;size_hints-&gt;height);</span>
 }
 
 #define MWM_HINTS_FLAGS_FIELD 0
<span class="hl kwa">-- </span>
2.41.0

</code></pre></td></tr></table>
</div> <!-- class=content -->
<div id="footer">
	<p>
		Copyright &copy; 2004-2023 aurweb Development Team &ndash;
		<strong>AUR packages are user produced content. Any use of the provided files is at your own risk.</strong>
	</p>
</div>
</div> <!-- id=cgit -->
</body>
</html>
